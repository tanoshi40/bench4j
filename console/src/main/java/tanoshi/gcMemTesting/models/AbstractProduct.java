package tanoshi.gcMemTesting.models;

public class AbstractProduct {
    private static int idCounter = 0;

    private final int price = (int) Math.round(Math.random());
    private final String id;

    public AbstractProduct() {
        id = "product-" + idCounter++;
    }

    public int getPrice() {
        return price;
    }

    public String getId() {
        return id;
    }
}
