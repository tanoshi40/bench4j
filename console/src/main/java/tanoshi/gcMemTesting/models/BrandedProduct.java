package tanoshi.gcMemTesting.models;

public class BrandedProduct extends AbstractProduct {
}
