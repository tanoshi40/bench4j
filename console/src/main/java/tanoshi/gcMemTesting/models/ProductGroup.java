package tanoshi.gcMemTesting.models;

import java.util.ArrayList;
import java.util.List;

public class ProductGroup {

    private List<AbstractProduct> products;

    public ProductGroup() {
        products = new ArrayList<>();
    }


    public void add(AbstractProduct product) {
        products.add(product);
    }

    public List<AbstractProduct> getProducts() {
        return products;
    }

    @Override
    public String toString() {
        return "{products=" + products.size() + '}';
    }
}
