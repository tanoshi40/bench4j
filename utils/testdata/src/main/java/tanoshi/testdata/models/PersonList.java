package tanoshi.testdata.models;

import java.util.List;

public final class PersonList {
    private final List<Person> persons;

    public PersonList(List<Person> persons) {
        this.persons = persons;
    }

    public List<Person> getPersons() {
        return persons;
    }
}
