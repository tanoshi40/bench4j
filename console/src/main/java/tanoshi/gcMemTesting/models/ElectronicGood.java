package tanoshi.gcMemTesting.models;

import tanoshi.gcMemTesting.models.AbstractProduct;

public class ElectronicGood extends AbstractProduct {
}
