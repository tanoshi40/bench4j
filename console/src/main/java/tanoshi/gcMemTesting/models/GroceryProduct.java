package tanoshi.gcMemTesting.models;

import tanoshi.gcMemTesting.models.AbstractProduct;

public class GroceryProduct extends AbstractProduct {
}
