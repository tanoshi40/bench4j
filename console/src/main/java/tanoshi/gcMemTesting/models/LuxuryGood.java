package tanoshi.gcMemTesting.models;

import tanoshi.gcMemTesting.models.AbstractProduct;

public class LuxuryGood extends AbstractProduct {
}
