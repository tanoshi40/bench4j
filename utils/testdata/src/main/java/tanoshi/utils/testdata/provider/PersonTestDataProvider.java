package tanoshi.utils.testdata.provider;

import tanoshi.utils.testdata.ITestDataProvider;
import tanoshi.utils.testdata.models.Person;

import java.util.List;

public class PersonTestDataProvider implements ITestDataProvider<List<Person>> {

    private final int testDataSize;
    private final List<Person> persons;

    public PersonTestDataProvider() { this(10); }
    public PersonTestDataProvider(int testDataSize) {
        System.out.println("init person data provider");
        this.testDataSize = testDataSize;
        this.persons = PersonProvider.getPersons(testDataSize);
        System.out.println("init person data provider done");
    }

    @Override
    public List<Person> getTestData() {
        return persons;
    }

    @Override
    public String getName() {
        return "PersonTestDataProvider-" + testDataSize;
    }

    public int getTestDataSize() {
        return testDataSize;
    }
}
