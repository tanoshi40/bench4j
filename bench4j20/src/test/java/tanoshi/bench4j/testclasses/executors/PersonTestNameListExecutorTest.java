package tanoshi.bench4j.testclasses.executors;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import tanoshi.testdata.models.PersonList;
import tanoshi.bench4j.testclasses.PersonTestDataProvider;

import static org.junit.jupiter.api.Assertions.*;
import static tanoshi.bench4j.testclasses.executors.PersonTestNameListExecutor.*;

@ExtendWith({SnapshotExtension.class})
class PersonTestNameListExecutorTest {
    private Expect expect;

    private static final PersonList persons = new PersonTestDataProvider(100).getTestData();

    @Test
    void getJoinedNamesStream() {
        String names = getStream(persons).namesString();
        expect.toMatchSnapshot(names);
    }

    @Test
    void getJoinedNamesStreamToListToStringJoin() {
        String names = getStreamToListToStringJoin(persons).namesString();
        expect.toMatchSnapshot(names);
    }

    @Test
    void getJoinedNamesStringBuilder() {
        String names = getStringBuilder(persons).namesString();
        expect.toMatchSnapshot(names);
    }

    @Test
    void getJoinedNamesStringJoin() {
        String names = getStringJoin(persons).namesString();
        expect.toMatchSnapshot(names);
    }

    @Test
    void getJoinedNamesStringJoinOptimized() {
        String names = getStringJoinOptimized(persons).namesString();
        expect.toMatchSnapshot(names);
    }

    @Test
    void getJoinedNamesStringJoiner() {
        String names = getStringJoiner(persons).namesString();
        expect.toMatchSnapshot(names);
    }

    @Test
    void getJoinedNameExecutorsAreEqual() {
        String stream = getStream(persons).namesString();
        String streamToList = getStreamToListToStringJoin(persons).namesString();
        String stringJoin = getStringJoin(persons).namesString();
        String stringBuilder = getStringBuilder(persons).namesString();
        String stringJoiner = getStringJoiner(persons).namesString();
        String stringJoinOptimized = getStringJoinOptimized(persons).namesString();

        assertEquals(stream, streamToList);
        assertEquals(stream, stringJoin);
        assertEquals(stream, stringBuilder);
        assertEquals(stream, stringJoiner);
        assertEquals(stream, stringJoinOptimized);
    }
}