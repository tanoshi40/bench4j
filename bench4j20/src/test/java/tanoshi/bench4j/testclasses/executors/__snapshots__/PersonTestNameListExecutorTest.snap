tanoshi.bench4j.testclasses.executors.PersonTestNameListExecutorTest.getJoinedNamesStream=[
phi, zeta, gamma, beta, kappa, rho, omega, sigma, lambda, beta, omikron, ypsilon, tau, rho, xi, tau, psi, lambda, kappa, kappa, chi, epsilon, lambda, iota, iota, zeta, gamma, rho, gamma, alpha, omikron, sigma, alpha, omega, delta, lambda, sigma, alpha, kappa, sigma, omikron, epsilon, mü, omikron, mü, lambda, zeta, phi, nü, rho, phi, rho, rho, lambda, beta, omega, beta, iota, omikron, phi, sigma, phi, eta, theta, pi, sigma, sigma, psi, theta, xi, chi, omega, chi, theta, zeta, mü, tau, beta, eta, zeta, phi, eta, epsilon, beta, tau, epsilon, kappa, xi, chi, mü, eta, lambda, kappa, epsilon, alpha, ypsilon, sigma, delta, epsilon, phi
]


tanoshi.bench4j.testclasses.executors.PersonTestNameListExecutorTest.getJoinedNamesStreamToListToStringJoin=[
phi, zeta, gamma, beta, kappa, rho, omega, sigma, lambda, beta, omikron, ypsilon, tau, rho, xi, tau, psi, lambda, kappa, kappa, chi, epsilon, lambda, iota, iota, zeta, gamma, rho, gamma, alpha, omikron, sigma, alpha, omega, delta, lambda, sigma, alpha, kappa, sigma, omikron, epsilon, mü, omikron, mü, lambda, zeta, phi, nü, rho, phi, rho, rho, lambda, beta, omega, beta, iota, omikron, phi, sigma, phi, eta, theta, pi, sigma, sigma, psi, theta, xi, chi, omega, chi, theta, zeta, mü, tau, beta, eta, zeta, phi, eta, epsilon, beta, tau, epsilon, kappa, xi, chi, mü, eta, lambda, kappa, epsilon, alpha, ypsilon, sigma, delta, epsilon, phi
]


tanoshi.bench4j.testclasses.executors.PersonTestNameListExecutorTest.getJoinedNamesStringBuilder=[
phi, zeta, gamma, beta, kappa, rho, omega, sigma, lambda, beta, omikron, ypsilon, tau, rho, xi, tau, psi, lambda, kappa, kappa, chi, epsilon, lambda, iota, iota, zeta, gamma, rho, gamma, alpha, omikron, sigma, alpha, omega, delta, lambda, sigma, alpha, kappa, sigma, omikron, epsilon, mü, omikron, mü, lambda, zeta, phi, nü, rho, phi, rho, rho, lambda, beta, omega, beta, iota, omikron, phi, sigma, phi, eta, theta, pi, sigma, sigma, psi, theta, xi, chi, omega, chi, theta, zeta, mü, tau, beta, eta, zeta, phi, eta, epsilon, beta, tau, epsilon, kappa, xi, chi, mü, eta, lambda, kappa, epsilon, alpha, ypsilon, sigma, delta, epsilon, phi
]


tanoshi.bench4j.testclasses.executors.PersonTestNameListExecutorTest.getJoinedNamesStringJoin=[
phi, zeta, gamma, beta, kappa, rho, omega, sigma, lambda, beta, omikron, ypsilon, tau, rho, xi, tau, psi, lambda, kappa, kappa, chi, epsilon, lambda, iota, iota, zeta, gamma, rho, gamma, alpha, omikron, sigma, alpha, omega, delta, lambda, sigma, alpha, kappa, sigma, omikron, epsilon, mü, omikron, mü, lambda, zeta, phi, nü, rho, phi, rho, rho, lambda, beta, omega, beta, iota, omikron, phi, sigma, phi, eta, theta, pi, sigma, sigma, psi, theta, xi, chi, omega, chi, theta, zeta, mü, tau, beta, eta, zeta, phi, eta, epsilon, beta, tau, epsilon, kappa, xi, chi, mü, eta, lambda, kappa, epsilon, alpha, ypsilon, sigma, delta, epsilon, phi
]


tanoshi.bench4j.testclasses.executors.PersonTestNameListExecutorTest.getJoinedNamesStringJoinOptimized=[
phi, zeta, gamma, beta, kappa, rho, omega, sigma, lambda, beta, omikron, ypsilon, tau, rho, xi, tau, psi, lambda, kappa, kappa, chi, epsilon, lambda, iota, iota, zeta, gamma, rho, gamma, alpha, omikron, sigma, alpha, omega, delta, lambda, sigma, alpha, kappa, sigma, omikron, epsilon, mü, omikron, mü, lambda, zeta, phi, nü, rho, phi, rho, rho, lambda, beta, omega, beta, iota, omikron, phi, sigma, phi, eta, theta, pi, sigma, sigma, psi, theta, xi, chi, omega, chi, theta, zeta, mü, tau, beta, eta, zeta, phi, eta, epsilon, beta, tau, epsilon, kappa, xi, chi, mü, eta, lambda, kappa, epsilon, alpha, ypsilon, sigma, delta, epsilon, phi
]


tanoshi.bench4j.testclasses.executors.PersonTestNameListExecutorTest.getJoinedNamesStringJoiner=[
phi, zeta, gamma, beta, kappa, rho, omega, sigma, lambda, beta, omikron, ypsilon, tau, rho, xi, tau, psi, lambda, kappa, kappa, chi, epsilon, lambda, iota, iota, zeta, gamma, rho, gamma, alpha, omikron, sigma, alpha, omega, delta, lambda, sigma, alpha, kappa, sigma, omikron, epsilon, mü, omikron, mü, lambda, zeta, phi, nü, rho, phi, rho, rho, lambda, beta, omega, beta, iota, omikron, phi, sigma, phi, eta, theta, pi, sigma, sigma, psi, theta, xi, chi, omega, chi, theta, zeta, mü, tau, beta, eta, zeta, phi, eta, epsilon, beta, tau, epsilon, kappa, xi, chi, mü, eta, lambda, kappa, epsilon, alpha, ypsilon, sigma, delta, epsilon, phi
]